roles/edpm_ovs/tasks/update.yml

```diff
@@ -0,0 +1,46 @@
+---
+- name: Check if ovs update is available
+  become: true
+  ansible.builtin.dnf:  # noqa: package-latest
+    name: "{{ edpm_ovs_packages }}"
+    state: latest
+  check_mode: true
+  register: ovs_update_check
+
+- name: Perform ovs update
+  when:
+    - ovs_update_check.rc == 0
+    - ovs_update_check.results | length > 0
+  become: true
+  block:
+    - name: Cache set ovs packages
+      ansible.builtin.include_tasks: download_cache.yml
+
+    - name: Update openvswitch packages without stopping the service
+      tags:
+        - update
+        - ovs
+      # Call dnf via the shell module as the ansible dnf module does not
+      # support passing --setopt options. This is required in order to disable any
+      # OVS rpm scriptlets which would stop and disable OVS services during
+      # a standard update. Further info:
+      # https://github.com/openstack-k8s-operators/dev-docs/blob/main/ovs-update.md#rpm-packaging
+      ansible.builtin.shell: >
+        dnf update -y {{ " ".join(edpm_ovs_packages) }} --cacheonly --setopt=tsflags=noscripts
```

Comment:
Let's handle empty list for edpm_ovs_packages as that could trigger full dnf update here, can add the condition on block itself above.

Reply:
This should be implicitly handled by the `Check if ovs update is available`. If you pass in an empty list there, it will result in an empty list of things to update. And because we have `ovs_update_check.results | length > 0` the update block will never run. But I can add an extra guard for this condition.

Reply:
Thanks, yes you are right, no need for extra check, it's fine i missed that part.